lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -335,30 +336,56 @@ class Courses(SysadminDashboardView):
     provides course listing information.
     """
 
-    def git_info_for_course(self, cdir):
+    def git_info_for_course(self, cdir, course_id):
         """This pulls out some git info like the last commit"""
-
-        cmd = ''
-        gdir = settings.DATA_DIR / cdir
         info = ['', '', '']
-
-        # Try the data dir, then try to find it in the git import dir
-        if not gdir.exists():
-            git_repo_dir = getattr(settings, 'GIT_REPO_DIR', git_import.DEFAULT_GIT_REPO_DIR)
-            gdir = path(git_repo_dir) / cdir
-            if not gdir.exists():
-                return info
-
-        cmd = ['git', 'log', '-1',
-               '--format=format:{ "commit": "%H", "author": "%an %ae", "date": "%ad"}', ]
-        try:
-            output_json = json.loads(subprocess.check_output(cmd, cwd=gdir))
-            info = [output_json['commit'],
-                    output_json['date'],
-                    output_json['author'], ]
-        except (ValueError, subprocess.CalledProcessError):
-            pass
-
+        # only if course has git import history
+        if CourseImportLog.objects.filter(course_id=course_id).count() > 0:
+            # check if course has git commit hash
+            logs = CourseImportLog.objects.filter(
+                course_id=course_id,
+                commit__ne=None
+            ).values_list(
+                'commit',
+                'date',
+                'author'
+            ).order_by('-created').first()
+
+            if logs:
+                info[0] = logs[0]
+                info[1] = logs[1]
+                info[2] = logs[2]
+            else:
+                # if course do not have git commit hash the import and cach.
```

Comment:
was this meant to be `courses` and `cache`?

Reply:
Can you fix the typos?

Reply:
it is single course, this function works for single course

Reply:
Done